### PR TITLE
enable parallel testing 🚀

### PR DESCRIPTION
**Pull request by nguyenvukhang**
each test can spawn its own database so now they can all run in parallel
BREAKING CHANGES:
`AppDataSource` is no longer supported. Import `db` from `database/src/config.ts` instead.
test it with `yarn test:all`

### CODE REVIEW

File: packages/database/package.json [hunk 1]
```diff
@@ -34,9 +34,9 @@
     "build": "tsc",
     "dev": "nodemon src/server.ts",
     "test": "yarn test:ci --detectOpenHandles",
-    "test:ci": "NODE_ENV=test jest --config ./tests/configs/ci.config.ts --runInBand",
-    "test:k": "NODE_ENV=test jest --config ./tests/configs/khang.config.ts --runInBand --detectOpenHandles",
-    "test:pull": "NODE_ENV=test jest --config ./tests/configs/pull.config.ts --runInBand --detectOpenHandles",
+    "test:ci": "NODE_ENV=test jest --config ./tests/configs/ci.config.ts",
+    "test:k": "NODE_ENV=test jest --config ./tests/configs/khang.config.ts --detectOpenHandles",
+    "test:pull": "NODE_ENV=test jest --config ./tests/configs/pull.config.ts --detectOpenHandles",
```
Comment (nguyenvukhang) [inline packages/database/package.json:39]: 🚀🚀🚀

File: packages/database/src/config.ts [hunk 2]
```diff
@@ -30,3 +31,17 @@ export const config = {
...
+  port: 3306,
+  username: config.username,
+  password: config.password,
+  database: config.database,
+  synchronize: true,
+  logging: false,
+  entities: config.entities,
+  migrations: config.migrations,
+  subscribers: [],
+})
```
Comment (nguyenvukhang) [inline packages/database/src/config.ts:47]: The new default database.
This replaces `AppDataSource`

File: packages/database/src/data-source.ts [hunk 3]
```diff
@@ -1,23 +1,9 @@
...
-  port: 3306,
-  username: config.username,
-  password: config.password,
-  database: config.database,
-  synchronize: true,
-  logging: false,
-  entities: config.entities,
-  migrations: config.migrations,
-  subscribers: [],
-})
```
Comment (nguyenvukhang) [inline packages/database/src/data-source.ts:19]: ❗️deprecated

File: packages/database/src/api/sql/reset/degree.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
 import { remove } from '../../../sql/remove'
+import { config } from '../../../config'
 
-remove.tables(['degree_modules_required_module', 'degree'])
+remove.tables(config.database, ['degree_modules_required_module', 'degree'])
```
Comment (weiseng18) [inline packages/database/src/api/sql/reset/degree.ts:4]: old table name

File: packages/database/src/api/sql/reset/module.ts [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { remove } from '../../../sql/remove'
+import { config } from '../../../config'
 
-remove.tables([
+remove.tables(config.database, [
   'user_modules_done_module',
   'user_modules_doing_module',
   'degree_modules_required_module',
```
Comment (weiseng18) [inline packages/database/src/api/sql/reset/module.ts:7]: old table name

File: packages/database/tests/module-pull.test.ts [hunk 6]
```diff
@@ -1,26 +1,30 @@
...
 }
 
 jest.setTimeout(60000)
 test('pull all modules from NUSMods', async () => {
-  await remove.tables([
+  await remove.tables(dbName, [
     'degree_modules_module',
     'user_modules_done_module',
     'user_modules_doing_module',
     'degree_modules_required_module',
```
Comment (weiseng18) [inline packages/database/tests/module-pull.test.ts:24]: degree_modules_required_module is deprecated

**Review by weiseng18 — APPROVED** [review]
lgtm, thanks for helping to fix the old/deprecated DB table names, and the tests are indeed fast

File: packages/database/src/repository/DAG.ts [hunk 7]
```diff
@@ -1,158 +1,206 @@
...
-  await container(async () => {
-    const retrieved = await DAGRepository.findOne({
-      where: {
-        id: dag.id,
-      },
-      relations: ['user', 'degree', 'modulesPlaced', 'modulesHidden'],
+      const [modulesPlaced, modulesHidden] = [
+        retrieved.modulesPlaced,
+        retrieved.modulesHidden,
+      ]
```
Comment (nguyenvukhang) [inline packages/database/src/repository/DAG.ts:141]: only used once

**Pull request merged**